src/gluonts/ext/hierarchicalforecast/_predictor.py

```diff
@@ -0,0 +1,309 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+from dataclasses import dataclass
+from typing import List, Optional, Union, Any, Dict
+
+import numpy as np
+import pandas as pd
+
+from statsforecast import StatsForecast
+from hierarchicalforecast.core import HierarchicalReconciliation
+from hierarchicalforecast.core import _build_fn_name
+
+from gluonts.core.component import validated
+from gluonts.dataset import DataEntry
+from gluonts.dataset.util import forecast_start
+from gluonts.model.predictor import RepresentablePredictor
+from gluonts.model.forecast import QuantileForecast
+from gluonts.ext.statsforecast import ModelConfig as _ModelConfig
+
+models_without_fitted_capability = [
+    "ADIDA",
+    "CrostonClassic",
+    "CrostonOptimized",
+    "CrostonSBA",
+    "IMAPA",
+    "SeasWA",
+    "TSB",
+    "WindowAverage",
+]
+
+
+def get_formatted_S(
+    _S: Union[List[List[int]], np.ndarray],
+    ts_names: List[Union[str, Any]],
+) -> pd.DataFrame:
+    S = np.array(_S)
+
+    return pd.DataFrame(
+        S, index=ts_names, columns=ts_names[S.shape[0] - S.shape[1] :]
```

Comment:
So it looks like we're just discarding the first few rows and never use them again (if I see through the code correctly), why then are the first rows required at all?

Reply:
All the rows are taken. What we do here is to take the matrix `S` and convert it to a suitable format for `hierarchicalforecast`, namely: make it a dataframe, set the index to `ts_names`, and provide names to the columns of `S`, which correspond to the first entries of `ts_names`. I will add a docstring to make it more accessible.